packages/ai-constructs/src/conversation/runtime/types.ts

```diff
@@ -24,4 +30,23 @@ export type ConversationTurnEvent = {
     };
   };
   messages: Array<ConversationMessage>;
+  toolsConfiguration?: {
+    tools: Array<{
+      name: string;
+      description: string;
+      inputSchema: ToolInputSchema;
+      graphqlRequestInputDescriptor: {
+        queryName: string;
+        selectionSet: string[];
+        propertyTypes: Record<string, string>;
+      };
+    }>;
+  };
+};
+
+export type ExecutableTool = {
```

Comment:
ExecutableTool & Tool array above seems to share common interface { name, description, input schema }, should we define this shared interface?

Reply:
Regarding this and https://github.com/aws-amplify/amplify-backend/pull/1830#discussion_r1712019497.

I have intentionally refrained from aggressively extracting named types in public api in order to not pollute IDE syntax highlighting. 

The reasoning behind this is that customer is not expected to create events themselves, yet these events are inputs  to lambda and one of public APIs so they have to have public type.


`ExecutableTool` and `Tool array` do share that commonality, but for the sake of simple `API.md` I did not seek common abstraction.
I was experimenting with `ToolSpecification` from bedrock as common type here https://github.com/aws-amplify/amplify-backend/pull/1830/commits/436a0a2a557bf444db231a5539130baff8ba321a#diff-a6be72429e1e423a62ed353b71d2e55bf9563168f5bb3e9f1cecb6be1eb9d2e2 , but it came with `string | undefined` typings which seemed unwanted for now.

Reply:
Got it, how are we tracking _intentionally refraining from_ exposing this type? should we leave a note if a type should not be refactored for that reason?

Reply:
> how are we tracking intentionally refraining from exposing this type?

We don't. Other than relying on natural instinct to doge naming types we can avoid naming (public api needs good names).

> should we leave a note if a type should not be refactored for that reason?

I will leave a note. Btw. this might change later if there's legitimate need to name more parts of the event, but this is going to be on demand activity later.

Reply:
added note